File: 300-longest-increasing-subsequence/longest-increasing-subsequence.cs
public class Solution {
    public int LengthOfLIS(int[] nums) {
        if (nums == null || nums.Length == 0) return 0;

        int[] dp = new int[nums.Length];
        int maxLength = 1;
        
        for (int i = 0; i < nums.Length; i++) {
            dp[i] = 1;
            for (int j = 0; j < i; j++) {
                if (nums[i] > nums[j]) {
                    dp[i] = Math.Max(dp[i], dp[j] + 1);
                }
            }
            maxLength = Math.Max(maxLength, dp[i]);
        }
        
        return maxLength;
    }
}

File: 300-longest-increasing-subsequence/longest-increasing-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var lengthOfLIS = function(nums) {
    let sub=[];
    for(let num of nums){
        let left=0, right=sub.length;
        while (left <right){
            let mid=Math.floor((left+right)/2);
            if(sub[mid]<num){left=mid+1;}
            else{right=mid;}
        }if(left<sub.length){sub[left]=num;}
        else{sub.push(num);}
    }return sub.length;
};